File: Hiking/Unity/Assets/PlayerScript.js
#pragma strict

public var stuckTo : boolean;
public var stuckStart : float; //this is when we started to stick to a platform
public var maxStuckTime : float = 3; // how long we're allowed to stay on a platform before falling
public var isJumping : boolean = false;
public var jumpTarget : GameObject;


var collisionObject : GameObject;
     

function Start () {
isJumping = false;
}

function Update () {


if(isJumping == true){
transform.position = Vector3.MoveTowards(transform.position,jumpTarget.transform.position, 1);
}


}//update



function jumpTo(target){
isJumping = true;
jumpTarget = target;
Debug.Log(jumpTarget.transform.position);
}

function jT(){
Debug.Log(isJumping);
}



    function OnCollisionEnter() {
   // if(collision.gameObject.tag=="Player"){
    Debug.Log("strike");
   // }
    }

File: Hiking/Unity/Assets/PlayerScript.cs
using UnityEngine;
using System.Collections;

public class PlayerScript : MonoBehaviour {

	public bool isStuck; //are we stuck to something?
	public float stuckStart = 0; //when did we start sticking?
	public float stuckTimeMax = 2; //how long are we allowed to stay stuck? (in seconds)
	public bool isJumping = false; //are we jumping to a new place?
	public bool isHooking = false; //have we thrown our hook?
	public Transform jumpTarget; //where are we jumping to?
	public GameObject hookTarget; //where are we hooking to?	
	public Transform playerTransform; //my transform
		
		
	// Use this for initialization
	void Start () {
	isJumping = false;
	}
	
	// Update is called once per frame
	void Update () {
	
		if(isJumping == true) //if we're jumping somewhere, move us to it each tick
		
		{
		
		Debug.Log(jumpTarget.transform.position);
	
		playerTransform.position = Vector3.MoveTowards(playerTransform.position,jumpTarget.position, (100*Time.deltaTime));
	
		}
			
	}
	
	public void JumpTo(Transform target)
	{
	
		jumpTarget = target;
		isJumping = true;
		Debug.Log(jumpTarget.position);
					
	}//jumpTo()
	
	
	
}
